app/src/main/java/fr/free/nrw/commons/upload/UploadMediaDetailAdapter.java

```diff
@@ -418,10 +419,45 @@ private void setUpRecentLanguagesSection(final List<Language> recentLanguages) {
                 languageHistoryListView.setAdapter(recentLanguagesAdapter);
             }
         }
+
+        /**
+         * Checks if the source string contains trailing whitespace
+         * @param source input string
+         * @return true if contains trailing whitespace and false otherwise
+         */
+        public Boolean checkTrailingWhitespace(String source) {
+            int len = source.length();
+            if (len == 0) {
+                return false;
+            }
+            return Character.isWhitespace(source.charAt(len - 1));
+        }
+
+        /**
+         * Removes any trailing whitespace from the source text.
+         * @param source input string
+         * @return a string without trailing whitespace
+         */
+        public String removeTrailingWhitespace(String source) {
+            while (checkTrailingWhitespace(source)) {
+                source = source.substring(0, source.length() - 1);
+            }
+            return source;
+        }
+
+        /**
+         * Convert Japanese spaces to English spaces
+         * @param source the source text
+         * @return a string with English space instead of Japanese space
+         */
+        public String convertJapSpaceToEngSpace(String source) {
```

Comment:
Could you please rename from `convertJapSpaceToEngSpace` to `convertIdeographicSpaceToLatinSpace`? Thanks!

Reply:
Done